### PR TITLE
frank/jit infra

### PR DESCRIPTION
**Pull request by soundsonacid**

### CODE REVIEW

File: src/marketmap.rs [hunk 1]
```diff
@@ -103,15 +102,16 @@ impl<T: AccountDeserialize + Clone + Send + Sync + Market + 'static> MarketMap<T
             self.sync().await?;
         }
 
-        if !self.subscribed.get() {
-            self.subscription.try_borrow_mut()?.subscribe::<T>().await?;
-            self.subscribed.set(true);
+        if !self.subscribed.load(Ordering::Relaxed) {
```
Comment (jordy25519) [inline src/marketmap.rs:105]: 💭 an idea to remove the synchronized subscribe bool, can encode state via generics
```
struct Subscribed;

impl <T> MarketMap<Subscribed, T> {
  /// subscribed state methods
}

struct Unsubscribed;

impl<T> MarketMap<Unsubscribed, T> {
  /// unsubscribed state methods
}
```

File: src/dlob/dlob_builder.rs [hunk 2]
```diff
@@ -26,16 +28,24 @@ impl DLOBBuilder {
...
-    pub async fn start_building(&mut self) -> SdkResult<()> {
-        self.slot_subscriber.subscribe().await?;
-        self.usermap.subscribe().await?;
-
-        self.build();
+    pub async fn start_building(builder: Arc<Mutex<Self>>) -> SdkResult<()> {
+        let mut locked_builder = builder.lock().await;
+        let rebuild_frequency = locked_builder.rebuild_frequency;
+        locked_builder.slot_subscriber.subscribe().await?;
+        locked_builder.usermap.subscribe().await?;
```
Comment (jordy25519) [inline src/dlob/dlob_builder.rs:35]: join

**Review by jordy25519 — COMMENTED** [review]
love your work

File: src/dlob/dlob_builder.rs (hunk 2 above)
Comment (soundsonacid) [reply src/dlob/dlob_builder.rs:35]: can't cause both require mutable borrow

File: src/user.rs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+};
+
+#[derive(Clone)]
+pub struct DriftUser {
+    pub pubkey: Pubkey,
+    subscription: WebsocketAccountSubscriber,
+    data_and_slot: Arc<RwLock<DataAndSlot<User>>>,
+}
+
+impl DriftUser {
```
[the remark this answers is not in the document]
Comment (soundsonacid) [reply src/user.rs:20]: i like this a lot

**Review by jordy25519 — APPROVED** [review]

**Comment by jordy25519** [comment]
just needs format, and tests passing - the 2 broken integration tests

**Pull request merged**